flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/ModelGenerator.java

```diff
@@ -117,9 +118,22 @@ public JsonElement getFlowModel(Document document, LineRange lineRange, Document
             }
         }
 
+        // Obtain the natural function names
+        Map<String, LineRange> naturalFunctions = new HashMap<>();
```

Comment:
Do we have a requirement to preserve the order when listing down the functions (e.g. insertion order or alphabetically)? If so we might need to use a LinkedHashMap or TreeMap here

Reply:
No, we don't need that. This is used to find the natural-function. The same kind of implementation we have in data-mappings.